src/Engine/AudioHandler/AudioHandler.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <map>
+
+#include "I_AudioHandler.h"
+
+class AudioHandler : public I_AudioHandler {
+public:
+    ~AudioHandler();
+    AudioHandler();
+    void init();
+    Mix_Music* load_music(std::string path);
```

Comment:
Can we make these smart unique pointers?

Reply:
This is actually a good idea. However I'll just take out load_music and make it a private function, and have my 
`std::map <std::string, Mix_Chunk*> sound_effects_;`
turn into 
`std::map <std::string, std::unique_ptr<Mix_Chunk*>> sound_effects_;`

That way, no function will ever need to have the Mix_Chunk. They just call `play_chunk`, and then I'll handle the rest internally and return errors accordingly. Thoughts? @enochtsang

Reply:
If other things call `play_chunk` what will they pass in to specify what chunk? A path?

Reply:
yea, they specify a path

Reply:
TODO: Tried to make this work today by keeping load and play. I think it's best if i just delete "load", and have the audio_handler::play() check if it exists and loads it if it doesn't. This avoids passing the unique_ptr from function to function.